src/view.c

```diff
@@ -156,7 +183,14 @@ view_moved(struct view *view)
 	assert(view);
 	wlr_scene_node_set_position(&view->scene_tree->node,
 		view->current.x, view->current.y);
-	view_discover_output(view);
+	/*
+	 * Only floating views change output when moved. Non-floating
+	 * views (maximized/tiled/fullscreen) are tied to a particular
+	 * output when they enter that state.
+	 */
+	if (view_is_floating(view)) {
+		view_discover_output(view);
+	}
```

Comment:
I didn't investigate this in detail yet, just wanted to point out that this *might* be an issue for interactive move from tiled state because we delay `view_set_untiled(view)`, I am not quite sure right now if `view_moved()` is called before the tiling state is reset.

Reply:
I've tested to/from tiled state with mouse/keyboard. Works fine.